shell/shell.go

```diff
@@ -72,9 +74,11 @@ func Run(ctx context.Context, exec Exec) (*ExecDetails, error) {
 		return nil, ctx.Oops().Wrap(err)
 	}
 
+	// Set to a non-nil empty slice to prevent access to current environment variables
+	cmd.Env = []string{}
 	if len(envParams.envs) != 0 {
 		ctx.Logger.V(6).Infof("using environment %s", logger.Pretty(envParams.envs))
-		cmd.Env = append(os.Environ(), envParams.envs...)
+		cmd.Env = append(cmd.Env, envParams.envs...)
```

Comment:
I think we still need these: HOME="$HOME" LC_CTYPE="${LC_ALL:-${LC_CTYPE:-$LANG}}" PATH="$PATH" USER="$USER" DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=$DOTNET_SYSTEM_GLOBALIZATION_INVARIANT
DEBIAN_FRONTEND=$DEBIAN_FRONTEND
PSModuleAnalysisCachePath=$PSModuleAnalysisCachePath
PS_VERSION=$PS_VERSION
CLOUDSDK_PYTHON=$CLOUDSDK_PYTHON
PS_INSTALL_FOLDER=$PS_INSTALL_FOLDER